spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/RabbitUtils.java

```diff
@@ -182,18 +182,18 @@ public static void closeMessageConsumer(Channel channel, Collection<String> cons
 		}
 	}
 
-	private static void cancel(Channel channel, String consumerTag) {
+	public static void cancel(Channel channel, String consumerTag) {
 		try {
 			channel.basicCancel(consumerTag);
 		}
-		catch (IOException e) {
-			if (logger.isDebugEnabled()) {
-				logger.debug("Error performing 'basicCancel'", e);
-			}
-		}
 		catch (AlreadyClosedException e) {
 			if (logger.isTraceEnabled()) {
-				logger.trace(channel + " is already closed");
+				logger.trace(channel + " is already closed", e);
+			}
+		}
+		catch (Exception e) {
+			if (logger.isDebugEnabled()) {
+				logger.debug("Error performing 'basicCancel'", e);
```

Comment:
We don't need a wrapping `if` - it is redundant by the reason of a constant for a message.